config.py

```diff
@@ -148,8 +148,8 @@ def get_cloud_init(name, base, password, provider, privateData, slavePortnum):
             )
             c['slaves'].append(slave)
     else:
-        for i, password in enumerate(slaveConfig['passwords']):
-            name = '%s-%d' % (base, i)
+        for index, password in enumerate(slaveConfig['passwords']):
```

Comment:
I suppose we could auto-generate the passwords for the non on-demand slaves too.